packages/icons/src/extract.js

```diff
@@ -2,20 +2,64 @@
 const fs = require('fs');
 const path = require('path');
 
-function extractFiles(folder) {
-	const dir = path.join(__dirname, folder);
+function getAbsolutePath(folder) {
+	if (folder.startsWith(__dirname)) {
+		return folder;
+	}
+	return path.join(__dirname, folder);
+}
 
+function getFiles(folder) {
+	const dir = getAbsolutePath(folder);
 	// case insensitive sort for windows users
-	return fs
-		.readdirSync(dir)
-		.sort(function(a, b) {
-			return a.localeCompare(b, { sensitivity: 'base' });
-		})
-		.reduce(
-			(state, file) =>
-				Object.assign(state, { [file.split('.svg')[0]]: fs.readFileSync(path.resolve(dir, file)) }),
-			{},
-		);
+	return fs.readdirSync(dir).sort((a, b) => a.localeCompare(b, { sensitivity: 'base' }));
+}
+
+function getIconId(file) {
+	return file.split('.svg')[0];
+}
+
+function assertUnique(files, acc) {
+	const alreadyExists = Object.keys(files).filter(file => {
+		const iconId = getIconId(file);
+		return acc[iconId];
+	});
+	if (alreadyExists.length > 0) {
+		throw new Error(`Icons already exists ${alreadyExists}`);
+	}
+}
+
+function extractFiles(folder) {
+	const dir = getAbsolutePath(folder);
+	return getFiles(folder).reduce((acc, file) => {
+		const p = path.resolve(dir, file);
+		if (fs.lstatSync(p).isDirectory()) {
+			const files = extractFiles(p);
+			// check if the same file exists in another bundle
+			assertUnique(files, acc);
+			return Object.assign(acc, files);
+		}
+		const iconId = getIconId(file);
```

Comment:
This code works if the folders are listed after the files.
But if they are mixed, it checks if the icons in subfolders are duplicates but not the icon files.

If you have a folder structure like that, it's ok
```
root
    |_ lol.svg
    |_ subFolder/
        |_ lol.svg
```

But this is ko
```
root
    |_ subFolder/
        |_ wtf.svg
    |_ wtf.svg
```
It will go over the subFolder, check if there are duplicates, then integrate the root > wtf.svg file, without checking.